modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -1408,15 +1410,30 @@ impl Wallet {
         change_tweak: &[u8; 33],
     ) -> Result<UnsignedTransaction, WalletOutputError> {
         match output {
-            WalletOutputV0::PegOut(peg_out) => self.offline_wallet().create_tx(
-                peg_out.amount,
-                peg_out.recipient.clone().assume_checked().script_pubkey(),
-                vec![],
-                self.available_utxos(dbtx).await,
-                peg_out.fees.fee_rate,
-                change_tweak,
-                None,
-            ),
+            WalletOutputV0::PegOut(peg_out) => {
+                let recipient = peg_out
+                    .recipient
+                    .clone()
+                    .require_network(self.cfg.consensus.network.0)
+                    .map_err(|_| {
+                        WalletOutputError::WrongNetwork(
+                            self.cfg.consensus.network.clone(),
+                            NetworkLegacyEncodingWrapper(get_network_for_address(
+                                &peg_out.recipient,
+                            )),
+                        )
+                    })?;
```

Comment:
This is a consensus change unless proven otherwise. We can forbid submitting withdrawal addresses with incorrect network, but forbidding them in consensus may break guardian recovery from a peer's history if someone already submitted such a transaction.

We could phase it in through a consensus version vote, something we already implemented in the wallet module anyway. cc @bradleystachurski @joschisan

Reply:
Ahh good catch! I wonder if this is worth a consensus change... I think it would make sense to drop a comment describing why we're calling `assume_checked()` and otherwise leave this code alone. We're eventually going to replace it with wallet v2 anyway.

Reply:
It's also more of a user foot-gun, for everything that matters we use script anyway, so nothing should blow up even if someone submits a wrong network address on purpose (the client won't do that on its own).

Reply:
Fixed! Also we already do reject transactions with the wrong network in `validate_tx()` which we call in `process_output()`:

https://github.com/fedimint/fedimint/blob/3b27f563b86ca3cd68e4f5b2399f9d10a79122bb/modules/fedimint-wallet-server/src/lib.rs#L713-L717